### ISSUE TITLE
Environment.from_existing_conda_environment fails with FileNotFoundError on existing conda environment

### ISSUE THREAD

**Issue by kato-m** [Azure/MachineLearningNotebooks#803]
I encounter the same problems as described in closed issue #721.  I am already on version 1.0.85, however. 

I've verified that the name passed as argument `conda_environment_name ` to `conda_environment_name`  is a valid conda environment.

Am I missing anything? Please find the technical details below.

Thanks.

```
> !python --version
Python 3.7.6

> !conda --version
conda 4.8.2

> !conda env list

# conda environments:
#
base                     C:\ProgramData\Miniconda3
experiments           *  C:\ProgramData\Miniconda3\envs\experiments

> import azureml.core
> print(azureml.core.VERSION)
1.0.85

> from azureml.core.environment import Environment
> env = Environment.from_existing_conda_environment(name = "experiments", conda_environment_name = "experiments")
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-25-356fdbee926e> in <module>
      1 from azureml.core.environment import Environment
----> 2 env = Environment.from_existing_conda_environment(name = "experiments", conda_environment_name = "experiments")
      3 #env.register(workspace=ws)

C:\ProgramData\Miniconda3\envs\experiments\lib\site-packages\azureml\core\environment.py in from_existing_conda_environment(name, conda_environment_name)
    630         """  # noqa: E501
    631         env = None
--> 632         if Environment._check_conda_installation() and Environment._check_conda_env_existance(conda_environment_name):
    633             try:
    634                 print("Exporting conda specifications for " +

C:\ProgramData\Miniconda3\envs\experiments\lib\site-packages\azureml\core\environment.py in _check_conda_installation()
    893         try:
    894             conda_version_cmd = ["conda", "--version"]
--> 895             conda_version = subprocess.check_output(conda_version_cmd).decode("UTF-8").replace("conda", "").strip()
    896 
    897             from distutils.version import LooseVersion

C:\ProgramData\Miniconda3\envs\experiments\lib\subprocess.py in check_output(timeout, *popenargs, **kwargs)
    409 
    410     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
--> 411                **kwargs).stdout
    412 
    413 

C:\ProgramData\Miniconda3\envs\experiments\lib\subprocess.py in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    486         kwargs['stderr'] = PIPE
    487 
--> 488     with Popen(*popenargs, **kwargs) as process:
    489         try:
    490             stdout, stderr = process.communicate(input, timeout=timeout)

C:\ProgramData\Miniconda3\envs\experiments\lib\subprocess.py in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors, text)
    798                                 c2pread, c2pwrite,
    799                                 errread, errwrite,
--> 800                                 restore_signals, start_new_session)
    801         except:
    802             # Cleanup if the child failed starting.

C:\ProgramData\Miniconda3\envs\experiments\lib\subprocess.py in _execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, unused_restore_signals, unused_start_new_session)
   1205                                          env,
   1206                                          os.fspath(cwd) if cwd is not None else None,
-> 1207                                          startupinfo)
   1208             finally:
   1209                 # Child is launched. Close the parent's copy of those pipe
FileNotFoundError: [WinError 2] Das System kann die angegebene Datei nicht finden
```

**Comment by vizhur** [comment]
@kato-m hm... interesting.. while we looking into issue can you please try these steps to mitigate the issue on your side
call either from subprocess (`export_existing_env_cmd = ["conda", "env", "export", "--no-builds", "--name", conda_environment_name]`) or just from cmd?
you can save conda output into a file and call
`env = Environment.from_conda_specification(name, conda_spec_file_name)`

**Comment by kato-m** [comment]
Thanks. It seems to work, although I was hoping, that this would upload local packages, too.

**Comment by vizhur** [comment]
That will create a specification what packages to install into the environment for your run execution. If we have no access to pull it from, like to the ones you installed in editable mode, we'll fail

**Comment by byronverz** [comment]
I have the same issue. 
`> !python --version
     3.7.9
> !conda --version
    4.9.2
> print(azureml.core.VERSION)
   1.18.0`

When I run `!conda env export --no-builds --name conda_environment_name` I get the environment dependencies as expected but not when using `Environment.from_existing_conda_environment`. I have checked the environment name as well but still get the error: `FileNotFoundError: [WinError 2] The system cannot find the file specified`.

**Comment by vizhur** [comment]
there is a known issue with jupyter stdout/err so subprocess.call doesn't work properly. you can follow this approach to unblock yourself 
https://github.com/Azure/MachineLearningNotebooks/issues/803#issuecomment-588412721

**Comment by anderl80** [comment]
Unfortunately I have same error while using vscode. Should I create another issue?

**Comment by vizhur** [comment]
You can export environment from terminal and instantiate environment from conda specification

**Comment by anderl80** [comment]
You mean `env = Environment.from_conda_specification(name="...", file_path="environment.yml")` and creating conda env manually in terminal?

Then I get the same error message this time hidden in azureml-logs/60_control_log.txt:

```
Running: ['cmd.exe', '/c', 'C:\\Users\\...\\AppData\\Local\\Temp\\azureml_runs\\proper-classification_1608386704_14b01e25\\azureml-environment-setup/conda_env_builder.bat']
Running: ['conda', '--version']
[WinError 2] Das System kann die angegebene Datei nicht finden
```

Conda is installed and works from command line.

**Comment by antonios4GitHub** [comment]
why is this closed?, it appears that the FileNotFoundError issue still exists when using jupyter or vs code notebook. The workaround solution works, but shouldn't it work without it after 2 years? Or is this issue encountered very few = neglectable times under very specific circumstances when using a notebook to submit a job?
